### ISSUE TITLE
can't reproduce the results

### ISSUE THREAD

**Issue by Interesting6** [mkl04/Semantic_Segmentation-Seismic_Images#1]
Thanks for your contribution! But I'm confused about the reproducing results. 

I just clone the repository and change the "root" variable to my dataset path in the code. Others remain the same. Then run the Baseline_UNet_Train.ipynb. I can't get the same result as yours.

<img width="542" alt="image" src="https://user-images.githubusercontent.com/22903497/156918586-48b5f7a3-a455-4c1c-8b8d-bb8de979d8ea.png">
here upper is my outputs, downer is origin output inside the ipynb.
<img width="779" alt="image" src="https://user-images.githubusercontent.com/22903497/156918607-2d858ccd-8263-43ed-b427-7395e14120c4.png">

**Comment by mkl04** [comment]
Hi! It is a baseline. To reproduce the paper results, try using the specific configuration (more epochs, new loss function, augmentation, number of filters, etc).

**Comment by Interesting6** [comment]
I have using the specific configuration (200 epochs, focal loss function, zoom augmentation, number of filters(32, 64, 128, 256, 512), lr=1e-4, 5epoch halve scheduler, ). But finally the results is
> Accuracy 1:  0.8254
> Accuracy 2:  0.8842
> Pixel Accuracy:  0.8525
> Mean Class Accuracy:  0.7567
> FWIOU:  0.7539
> Class Accuracy:  [0.342 0.769 0.706 0.85  0.91  0.963]
> mIoU:  0.6214679027270419

**Comment by Interesting6** [comment]
besides, what's the purpose of alpha insight your focalloss implementation?
` python
def categorical_focal_loss(gamma=2., alpha=1): 
    def categorical_focal_loss_fixed(y_true, y_pred):
        y_pred /= tf.math.reduce_sum(y_pred, axis=-1, keepdims=True)
        y_pred = tf.clip_by_value(y_pred, K.epsilon(), 1. - K.epsilon())
        cross_entropy = - y_true * tf.math.log(y_pred)
        loss = alpha * tf.math.pow(1 - y_pred, gamma) * cross_entropy
        return tf.math.reduce_mean(loss, -1)
    return categorical_focal_loss_fixed
`
In theory, alpha should be classweight. However your 'alpha' acts a scalar quantity actually, I think.

**Comment by mkl04** [comment]
I'm not sure why your results are like that. I did a test with only 5 epochs and got the following results:

![image](https://user-images.githubusercontent.com/41549216/159395311-8a639adb-fa39-4907-b65f-b4f50eac29fb.png)


It may be because of the version of tensorflow/keras. I used to have different results when training at Colab compared to the local. I'm going to update the repository with a requirements.txt

Then, about your second question, I think this issue could help you:
https://github.com/umbertogriffo/focal-loss-keras/issues/1
